### DISCUSSION TITLE
Use Latest version of a GitHub Action

### DISCUSSION THREAD

**Discussion by alvinsjoy** [community/community#88510] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi is it possible to use the latest version of a published GitHub Action from the Marketplace in a workflow?

This would be nice if it was somehow possible:

 ```yml
- name: Deploy to GitHub Pages
   uses: crazy-max/ghaction-github-pages@latest
```

For example, with a npm package you can do `npm install node@latest` to use the latest version of that package.

On an action's page, when you click on "Use latest version", it shows the following dialogue, where the version number is hard-coded:
![image](https://github.com/community/community/assets/89687635/8235f8cd-b056-401b-b795-b08c977c6994)

**Comment by KK-Repos** [comment, marked as answer]
Hey @alvinsjoy 

You can use the "**master**" or "**main**" branch instead of a tag to automatically get the latest changes.
Example:

```
   - name: Checkout
       uses: actions/checkout@master
```

Testing with the master branch for checkout actions has been successful. For your reference, please see the results at test-checkout: https://github.com/KK-Repos/Actions-Lab/actions/runs/7591812055/job/20680378961

 💡 Third-party action suggestion:

[GitHub Actions Version Updater](https://github.com/marketplace/actions/github-actions-version-updater) automatically finds and updates outdated actions in workflows.


:rotating_light:  It is recommended to use a tag for stability. Using a tag ensures that you are using a specific, tested version of the action, preventing unexpected failures.

**Reply by dirkf** [reply]
OP asked for the "latest version". What is the solution if that is interpreted as the "latest released version" rather than whatever happens to be the latest code in some branch of the action's repo?

**Reply by mi-hol** [reply]
@KK-Repos what is the actual tag name referencing "latest released version" of an action?

If I trust comments on StackOverflow, there seems to be no reliable way to achieve just that.

**Comment by Evander-Greer** [comment]
GitHub Actions does not currently support using @latest directly in the workflow YAML file. Instead, you need to specify the exact version of the action. The versioning for GitHub Actions is done through tags, which are hard-coded into the action’s YAML file when added to the workflow.

You cannot dynamically fetch the latest version of an action the way you can with npm packages. Instead, the best practice is to:

Use a specific version tag, like @v4.0.0.

Manually update the version in your workflow as new versions are released.

Suggested Workflow Example:
- name: Deploy to GitHub Pages
  uses: crazy-max/ghaction-github-pages@v4.0.0

Why Is This the Case?

The GitHub Actions ecosystem is designed to ensure that workflows are reproducible and stable. By specifying exact versions, you ensure that the action behaves consistently across different runs. Allowing @latest could introduce issues if an action changes behavior in a new release, potentially breaking workflows.

Final Note:

If you'd like to stay on top of new releases, you can regularly check the GitHub Action's releases page or use tools like Dependabot to notify you when a new version of a GitHub Action is available.
